src/modules/mana/sagas.ts

```diff
@@ -484,6 +471,120 @@ function* handleConnectWalletSuccess(_action: ConnectWalletSuccessAction) {
   }
 }
 
+function formatImportWithdrawalError(msg: string) {
+  return `${IMPORT_WITHDRAWAL_REQUEST} - ${msg}`
+}
+
+export enum ImportWithdrawalErrors {
+  NOT_FOUND = 'notFound',
+  NOT_WITHDRAWAL = 'notWithdrawal',
+  NOT_OWN_TRANSACTION = 'notOwnTransaction',
+  ALREADY_PROCESSED = 'alreadyProcessed',
+}
+
+export const importWithdrawalErrors = {
+  notFound: formatImportWithdrawalError(ImportWithdrawalErrors.NOT_FOUND),
+  notWithdrawal: formatImportWithdrawalError(
+    ImportWithdrawalErrors.NOT_WITHDRAWAL
+  ),
+  notOwnTransaction: formatImportWithdrawalError(
+    ImportWithdrawalErrors.NOT_OWN_TRANSACTION
+  ),
+  alreadyProcessed: formatImportWithdrawalError(
+    ImportWithdrawalErrors.ALREADY_PROCESSED
+  ),
+  other: (msg: string) => formatImportWithdrawalError(msg),
+}
+
+function* handleImportWithdrawalRequest(action: ImportWithdrawalRequestAction) {
+  const {
+    payload: { txHash },
+  } = action
+
+  try {
+    const address: string | undefined = yield select(getAddress)
+    const networks: ReturnType<typeof getNetworks> = yield select(getNetworks)
+    const provider: Provider = yield call(() =>
+      getNetworkProvider(networks![Network.MATIC].chainId)
```

Comment:
if you want you can avoid one `yield` by using the helper `getChainIdByNetwork(Network.MATIC)` from `decentraland-dapps/dist/lib/eth`